File: 05 - Formação React/01 - Programar com JS Orientado a Objetos/02 - JS - Interfaces e Herança em Orientação a Objetos/Funcionarios/Funcionario.js
class Funcionario{
    constructor(nome, salario, cpf){
        this._nome = nome;
        this._salario = salario;
        this._cpf = cpf;

        this.bonificacao = 1;
        this._senha;
    }

    cadastraSenha(senha){
        this._senha = senha;
    }

    autenticar(senha){
        return senha === this._senha;
    }
}

export default Funcionario;

File: 14 - Formação C# e POO/03 - Usando herança e implementando interfaces/bytebank_ADM/Funcionarios/Funcionario.cs
namespace bytebank_ADM.Funcionarios{
    public abstract class Funcionario{
        public string Nome { get; set; } = "";
        public string Cpf { get; private set; } = "";
        public double Salario { get; protected set; } = 0;
        public static int TotalDeFuncionarios { get; private set; }
        
        public int tipo = 0;

        public Funcionario(string cpf, double salario) {
            this.Cpf = cpf;
            this.Salario = salario;
            TotalDeFuncionarios++;
        }

        public abstract double GetBonificacao();
        
        public abstract void AumentarSalario();

    }
}